File: my-app/refactor_dest.js
import fs from 'fs';

const filePath = './src/pages/DestinationTemplate.tsx';
let content = fs.readFileSync(filePath, 'utf8');

// The lines we want to remove are between "// ============= INTERNAL COMPONENTS ============="
// and "// ============= MAIN TEMPLATE ============="

const startToken = "// ============= INTERNAL COMPONENTS =============";
const endToken = "// ============= MAIN TEMPLATE =============";

const startIndex = content.indexOf(startToken);
const endIndex = content.indexOf(endToken);

if (startIndex !== -1 && endIndex !== -1) {
    const before = content.substring(0, startIndex);
    const after = content.substring(endIndex);
    
    // Add imports
    const imports = `import { BookingForm } from "../components/BookingForm";\nimport { Card, ImgCard, PlaceCard, Section } from "../components/DestinationUI";\n\n`;
    
    const newContent = before + imports + after;
    fs.writeFileSync(filePath, newContent, 'utf8');
    console.log("Successfully removed internal components and added imports!");
} else {
    console.log("Tokens not found.");
}

File: my-app/refactor_dest.py
import sys

file_path = './src/pages/DestinationTemplate.tsx'

with open(file_path, 'r') as f:
    content = f.read()

start_token = "// ============= INTERNAL COMPONENTS ============="
end_token = "// ============= MAIN TEMPLATE ============="

start_idx = content.find(start_token)
end_idx = content.find(end_token)

if start_idx != -1 and end_idx != -1:
    before = content[:start_idx]
    after = content[end_idx:]
    imports = 'import { BookingForm } from "../components/BookingForm";\nimport { Card, ImgCard, PlaceCard, Section } from "../components/DestinationUI";\n\n'
    new_content = before + imports + after
    with open(file_path, 'w') as f:
        f.write(new_content)
    print("Success")
else:
    print("Tokens not found")
